lib/bundler/dsl.rb

```diff
@@ -34,7 +34,9 @@ def initialize
     end
 
     def eval_gemfile(gemfile, contents = nil)
-      @gemfile = Pathname.new(gemfile)
+      expanded_gemfile_path = Pathname.new(gemfile).expand_path
+      @original_gemfile ||= expanded_gemfile_path
```

Comment:
Initially, I thought that we wanted to restore `@gemfile` to the original, top-level `Gemfile` path after each `eval_gemfile` invocation. However, you're right, I think if we want to support multiple, recursive nesting of `eval_gemfile` invocations, we have to restore `@gemfile` to the Gemfile path specified for the recursive level above each `eval_gemfile` call. This way, if people are doing things like chaining > 2 Gemfiles and `eval_gemfile` calls (I hope not though...), `@gemfile` is set properly on each recursive level.